kasa/aestransport.py

```diff
@@ -132,55 +118,70 @@ async def client_post(self, url, params=None, data=None, json=None, headers=None
 
         return resp.status_code, response_data
 
-    async def send_secure_passthrough(self, request):
+    async def send_secure_passthrough(self, request: str):
         """Send encrypted message as passthrough."""
         url = f"http://{self.host}/app"
         if self.login_token:
             url += f"?token={self.login_token}"
-        raw_request = json_dumps(request)
-        encrypted_payload = self.encryption_session.encrypt(raw_request.encode())
+
+        encrypted_payload = self.encryption_session.encrypt(request.encode())  # type: ignore
         passthrough_request = {
             "method": "securePassthrough",
             "params": {"request": encrypted_payload.decode()},
         }
         status_code, resp_dict = await self.client_post(url, json=passthrough_request)
+        _LOGGER.debug(f"secure_passthrough response is {status_code}: {resp_dict}")
         if status_code == 200 and resp_dict["error_code"] == 0:
-            response = self.encryption_session.decrypt(
+            response = self.encryption_session.decrypt(  # type: ignore
                 resp_dict["result"]["response"].encode()
             )
+            _LOGGER.debug(f"decrypted secure_passthrough response is {response}")
             resp_dict = json_loads(response)
-            if resp_dict["error_code"] != 0:
-                raise SmartDeviceException(
-                    f"Could not complete send, response was {resp_dict}",
-                )
-            if "result" in resp_dict:
-                return resp_dict["result"]
+            return resp_dict
         else:
+            self.handshake_done = False
+            self.login_token = None
             raise AuthenticationException("Could not complete send")
 
-    def get_aes_request(self, method, params=None):
-        """Get a request message."""
-        request = {
-            "method": method,
-            "params": params,
-            "requestID": self.request_id_generator.generate_id(),
-            "request_time_milis": round(time.time() * 1000),
-            "terminal_uuid": self.terminal_uuid,
-        }
-        return request
-
-    async def perform_login(self, login_v2):
+    async def perform_login(self, login_request, login_v2):
         """Login to the device."""
         self.login_token = None
 
+        if isinstance(login_request, str):
+            login_request = json_loads(login_request)
```

Comment:
I seem to recall I was originally trying to use dicts but it got complicated because `transport.send()` has to take a string because the `SmartProtocol.query()` takes a method name rather than a dict.  We could probably improve this down the line so we have a more concrete implementation of a `query_request`.